contracts/actions/convex/helpers/ConvexHelper.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import "../../../interfaces/convex/IConvexToken.sol";
+import "../../../interfaces/convex/IConvexToken.sol";
+import "../../../interfaces/convex/IBaseRewardPool.sol";
+import "../../../interfaces/convex/IRewardPool.sol";
+import "../../../utils/SafeMath.sol";
+import "../../../utils/TokenUtils.sol";
+
+import "./MainnetConvexAddresses.sol";
+
+contract ConvexHelper is MainnetConvexAddresses {
+    using SafeMath for uint256;
+    using TokenUtils for address;
+
+    uint256 internal immutable CVX_REDUCTION_PER_CLIFF;
+    uint256 internal immutable CVX_TOTAL_CLIFFS;
+    uint256 internal immutable CVX_MAX_SUPPLY;
+
+    struct Reward {
+        address token;
+        uint256 earned;
+    }
+
+    enum DepositOption {
+        WRAP,
+        STAKE,
+        WRAP_AND_STAKE
+    }
+
+    enum WithdrawOption {
+        UNWRAP,
+        UNSTAKE,
+        UNSTAKE_AND_UNWRAP
+    }
+
+    constructor() {
+        IConvexToken CVX = IConvexToken(CVX_ADDR);
+        CVX_REDUCTION_PER_CLIFF = CVX.reductionPerCliff();
+        CVX_TOTAL_CLIFFS = CVX.totalCliffs();
+        CVX_MAX_SUPPLY = CVX.maxSupply();
+    }
+
+    /// @dev taken from Cvx.mint()
+    function cvxMintAmount(uint256 _amount) public view returns (uint256) {
+        uint256 supply = IERC20(CVX_ADDR).totalSupply();
+        
+        //use current supply to gauge cliff
+        //this will cause a bit of overflow into the next cliff range
+        //but should be within reasonable levels.
+        //requires a max supply check though
+        uint256 cliff = supply.div(CVX_REDUCTION_PER_CLIFF);
+        //mint if below total cliffs
+        uint256 totalCliffs = CVX_TOTAL_CLIFFS;
+        if(cliff < totalCliffs){
+            //for reduction% take inverse of current cliff
+            uint256 reduction = totalCliffs.sub(cliff);
+            //reduce
+            _amount = _amount.mul(reduction).div(totalCliffs);
+
+            //supply cap check
+            uint256 amtTillMax = CVX_MAX_SUPPLY.sub(supply);
+            if(_amount > amtTillMax){
+                _amount = amtTillMax;
+            }
+        }
+
+        return _amount;
+    }
+
+    function earnedRewards(address _account, address _rewardContract) public view returns (
```

Comment:
Add comments how it might leave empty 0x0 add the end if there are crv/cvx addresses